### PR TITLE
Update repository

### LINKED ISSUE #1: Add factories
There's nothing that can create events or event managers.

Add factory interfaces.

### LINKED ISSUE #2: Update repository
Metadata from the original clone needs to be updated.

### PR DESCRIPTION
**Pull request by XedinUnknown**
- Converted to Dhii standards.
- Added factories for both `EventInterface` and `EventManagerInterface`.
- `EventManagerInterface` no longer extends PSR interface, and is strict, yet forward-compatible.
- Tests added.
- BC broken.

Fixes #3.  
Fixes #2.  
Fixes #1.

### CODE REVIEW

**Review by mecha — CHANGES_REQUESTED** [review]
Looks good. Just missing the tests for the factory interfaces.

File: .idea/php.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
   <component name="PhpProjectSharedConfiguration" php_language_level="5.3.0" />
   <component name="PhpUnit">
     <phpunit_settings>
-      <PhpUnitSettings load_method="CUSTOM_LOADER" configuration_file_path="D:/projects/dhii/event-manager-interface/phpunit.xml" custom_loader_path="D:/projects/dhii/event-manager-interface/vendor/autoload.php" phpunit_phar_path="" use_configuration_file="true" />
+      <PhpUnitSettings load_method="CUSTOM_LOADER" custom_loader_path="F:/projects/dhii/event-interface/vendor/autoload.php" phpunit_phar_path="" />
```
Comment (mecha) [inline .idea/php.xml]: Why is it that this config entry never uses the `$PROJECT_DIR$` variable for you?

File: .idea/php.xml [hunk 2]
```diff
@@ -37,12 +37,16 @@
...
+      <path value="$PROJECT_DIR$/vendor/dhii/stringable-interface" />
+      <path value="$PROJECT_DIR$/vendor/dhii/exception-interface" />
+      <path value="$PROJECT_DIR$/vendor/dhii/factory-interface" />
     </include_path>
   </component>
   <component name="PhpProjectSharedConfiguration" php_language_level="5.3.0" />
   <component name="PhpUnit">
     <phpunit_settings>
-      <PhpUnitSettings load_method="CUSTOM_LOADER" configuration_file_path="F:/projects/dhii/psr-14/phpunit.xml" custom_loader_path="F:/projects/dhii/psr-14/vendor/autoload.php" phpunit_phar_path="" use_configuration_file="true" />
+      <PhpUnitSettings load_method="CUSTOM_LOADER" custom_loader_path="F:/projects/dhii/event-interface/vendor/autoload.php" phpunit_phar_path="" />
```
Comment (mecha) [inline .idea/php.xml:49]: Why is it that this config entry never uses the `$PROJECT_DIR$` variable for you?

Comment (XedinUnknown) [reply .idea/php.xml:49]: Mystery. Does it for you?

Comment (mecha) [reply .idea/php.xml:49]: Yes. In fact when I clone projects that have paths like this, after changing the beginning of the path from `F:\projects` to my `D:\dev`, PHPS automatically changes it to `$PROJECT_DIR$`.

Comment (XedinUnknown) [reply .idea/php.xml:49]: I would love my configs to do that. But they don't. Do you remember what it was like before the PHPS update?

Comment (mecha) [reply .idea/php.xml:49]: No. Would have to go look at some projects that I had created at the time.

**Pull request merged**